### PR TITLE
feat: bring back walletconnect v1

### PR DESCRIPTION
**Pull request by grabbou**
Initially, we deprecated WalletConnect V1 and replaced it immediately with V2. We have made decision internally that we're going to ship both packages side by side till V1 is deprecated, instead of bumping major version and cutting legacy branch. This PR brings back V1. 

Changelog:
- dropping peer dependencies in favour of dependencies so both WC can run side by side
- added `@walletconnect/types` dependency that was missing before inside `walletconnect` (v1) package
- bumped version locally to latest published to `npm` (which is currently v2), next publish will bring downgrade `@web3-react/walletconnect` back to v1

Screenshot 2023-02-28 at 20 12 14

PS. I did `git mv` to move `walletConnectCard.tsx` to `walletConnectV2Card.tsx` but unfortunately, that didn't help with the diff - it still reports that I did changes in place and created a new `walletConnectV2Card.tsx`, which is not really what happened.

### CODE REVIEW

File: packages/walletconnect/package.json
Comment (zzmp) [inline packages/walletconnect/package.json:29]: nit: this should be sorted first in the dependencies list

File: packages/walletconnect/package.json
Comment (zzmp) [inline packages/walletconnect/package.json:33]: nit: this should be sorted first in the devDependencies list

File: example/components/ConnectWithSelect.tsx
Comment (zzmp) [inline example/components/ConnectWithSelect.tsx:91]: nit: alphabetize

File: example/components/ProviderExample.tsx
Comment (zzmp) [inline example/components/ProviderExample.tsx:1]: nit: just import, don't use `type`, because it's a maintenance burden to upkeep

File: example/next.config.js
Comment (zzmp) [inline example/next.config.js:11]: If you've root-caused this, can you be more explicit in the comment as to what is actually going wrong?

File: example/package.json
Comment (zzmp) [inline example/package.json:23]: nit: this should go above v2 alphabetically

File: example/components/connectorCards/WalletConnectV2Card.tsx
Comment (zzmp) [inline example/components/connectorCards/WalletConnectV2Card.tsx:12]:
```suggestion
export default function WalletConnectV2Card() {
```

**Review by zzmp — APPROVED** [review]
mostly nits, but I would like to see the todo updated to be more explicit about why `loose` is necessary.

File: example/next.config.js
Comment (grabbou) [reply example/next.config.js:11]: I am still waiting for the response from WalletConnect team to provide further explanation and it's primary blocker to merge this PR. Don't want to ship a workaround w/o fully understanding what is going on here.

**Comment by grabbou** [comment]
Thanks for the review @zzmp! I do agree with you on the `loose` workaround and I am waiting for further context before making it more explicit and then, merging this PR!

File: example/components/ProviderExample.tsx
Comment (grabbou) [reply example/components/ProviderExample.tsx:1]: Personally, I prefer the explicit use of `type` where I am requiring a specific package for type definitions only. However, it is correct that TypeScript removes imports not utilized as values by default (https://www.typescriptlang.org/tsconfig#importsNotUsedAsValues), so I guess this nit makes sense.

**Comment by grabbou** [comment]
Unit test failure due to network failure, unrelated.

**Pull request merged**
